components/sup/src/http_gateway.rs

```diff
@@ -244,6 +259,32 @@ fn health(req: &mut Request) -> IronResult<Response> {
     }
 }
 
+fn service(req: &mut Request) -> IronResult<Response> {
+    let state = req.get::<persistent::Read<ManagerFs>>().unwrap();
+    let service_group = match build_service_group(req) {
+        Ok(sg) => sg,
+        Err(_) => return Ok(Response::with(status::BadRequest)),
+    };
+    match File::open(&state.services_data_path) {
```

Comment:
Maybe consider a `service_from_file` function to dry up the service deserialization but not a show stopper.